_posts/blog/2022-12-1-vector-library-vs-vector-database.md

```diff
@@ -85,14 +83,14 @@ The table below summarizes the differences between vector libraries and database
 | Incremental importing, concurrent reading while importing| No (some do, e.g. hnswlib) | Yes |
 | Stores objects and vectors | No | Yes|
 |Speed | Typically faster than full-blown database|  Typically slower than pure library|
-| Performance optimized for | In-memory similarity search| End2end-callstack, including: vector search, object retrieval from persistent storage, optional inverted index filtering, network requests, etc. |
+| Performance optimized for | In-memory similarity search| Full functionality, including: vector search, object retrieval from persistent storage, optional inverted index filtering, etc. |
```

Comment:
I would leave it as is. This table was done by Etienne and not sure switching end2end-callstack to full functionality is right.

Reply:
@etiennedi: would it make sense to use a more common term here? We're the only ones on the web that use "end2end-callstack" :)

<img width="731" alt="image" src="https://user-images.githubusercontent.com/33569/206022605-ae39d202-b7e5-4175-b17a-e54a59564dac.png">

Reply:
Of course, feel free to pick a better term. Now I'm almost proud of picking something that is so unique that no one uses it 😅

My motivation behind that term was basically to say: With a library you have a single "function", like one moving gear inside a big machine. With the database the level of abstraction is so much higher that you get everything, including that one specific function. The gear is considerably more efficient at doing gear-y stuff, but if you need a full lawn mower it's less important how efficient a single gear is, if for example the blades are dull.

Reply:
My suggestion was "full functionality". What do you think @etiennedi ?

Reply:
Works for me :-)